### PR TITLE
Updates atmosphere version.

### LINKED ISSUE #411: Update Atmosphere library to the last version
For our [RestAPI](https://github.com/freedomotic/freedomotic/tree/master/plugins/devices/restapi-v3) plugin we are using [Atmosphere](https://github.com/Atmosphere/atmosphere) framework.
Currently it uses 2.2.5 version but the last available is 2.4.14.
We should update the code.

### PR DESCRIPTION
**Pull request by dmitraver**
Closes #411

### CODE REVIEW

**Comment by mcicolella** [comment]
Hi @dmitraver have you run the software with your changes?
I have many errors

`
2018-10-03 10:34:11,792 WARN  [main] - Error while loading all plugins. Impossible to load "null" due to "com.freedomotic.exceptions.PluginLoadingException: Generic error while loading boundle /home/emmecilab/Scrivania/ultimo/plugins/devices/restapi-v3/restapi-v3-3.1.jar
	at com.freedomotic.plugins.impl.BoundleLoaderDevices.loadBoundle(BoundleLoaderDevices.java:70)
	at com.freedomotic.plugins.impl.PluginsManagerImpl.loadSingleBundle(PluginsManagerImpl.java:144)
	at com.freedomotic.plugins.impl.PluginsManagerImpl.loadAllPlugins(PluginsManagerImpl.java:125)
	at com.freedomotic.app.Freedomotic.start(Freedomotic.java:259)
	at com.freedomotic.app.Freedomotic.main(Freedomotic.java:456)
Caused by: com.google.inject.ConfigurationException: Guice configuration errors:

1) No implementation for org.atmosphere.cpr.BroadcasterFactory was bound.
  while locating org.atmosphere.cpr.BroadcasterFactory
    for field at com.freedomotic.plugins.devices.restapiv3.resources.atmosphere.AtmosphereEventResource.factory(AtmosphereEventResource.java:57)
  while locating com.freedomotic.plugins.devices.restapiv3.resources.atmosphere.AtmosphereEventResource
    for field at com.freedomotic.plugins.devices.restapiv3.RestAPIv3.atmosphereEventResource(RestAPIv3.java:43)
  while locating com.freedomotic.plugins.devices.restapiv3.RestAPIv3

2) No implementation for org.atmosphere.cpr.BroadcasterFactory was bound.
  while locating org.atmosphere.cpr.BroadcasterFactory
    for field at com.freedomotic.plugins.devices.restapiv3.resources.atmosphere.AtmosphereMessageCalloutResource.factory(AtmosphereMessageCalloutResource.java:47)
  while locating com.freedomotic.plugins.devices.restapiv3.resources.atmosphere.AtmosphereMessageCalloutResource
    for field at com.freedomotic.plugins.devices.restapiv3.RestAPIv3.atmosphereMessageCalloutResource(RestAPIv3.java:43)
  while locating com.freedomotic.plugins.devices.restapiv3.RestAPIv3

3) No implementation for org.atmosphere.cpr.BroadcasterFactory was bound.
  while locating org.atmosphere.cpr.BroadcasterFactory
    for field at com.freedomotic.plugins.devices.restapiv3.resources.atmosphere.AtmosphereObjectChangeResource.factory(AtmosphereObjectChangeResource.java:47)
  while locating com.freedomotic.plugins.devices.restapiv3.resources.atmosphere.AtmosphereObjectChangeResource
    for field at com.freedomotic.plugins.devices.restapiv3.RestAPIv3.atmosphereObjectChangeResource(RestAPIv3.java:43)
  while locating com.freedomotic.plugins.devices.restapiv3.RestAPIv3

4) No implementation for org.atmosphere.cpr.BroadcasterFactory was bound.
  while locating org.atmosphere.cpr.BroadcasterFactory
    for field at com.freedomotic.plugins.devices.restapiv3.resources.atmosphere.AtmospherePluginChangeResource.factory(AtmospherePluginChangeResource.java:48)
  while locating com.freedomotic.plugins.devices.restapiv3.resources.atmosphere.AtmospherePluginChangeResource
    for field at com.freedomotic.plugins.devices.restapiv3.RestAPIv3.atmospherePluginChangeResource(RestAPIv3.java:43)
  while locating com.freedomotic.plugins.devices.restapiv3.RestAPIv3

5) No implementation for org.atmosphere.cpr.BroadcasterFactory was bound.
  while locating org.atmosphere.cpr.BroadcasterFactory
    for field at com.freedomotic.plugins.devices.restapiv3.resources.atmosphere.AtmosphereZoneChangeResource.factory(AtmosphereZoneChangeResource.java:48)
  while locating com.freedomotic.plugins.devices.restapiv3.resources.atmosphere.AtmosphereZoneChangeResource
    for field at com.freedomotic.plugins.devices.restapiv3.RestAPIv3.atmosphereZoneChangeResource(RestAPIv3.java:43)
  while locating com.freedomotic.plugins.devices.restapiv3.RestAPIv3

5 errors
	at com.google.inject.internal.InjectorImpl.getMembersInjector(InjectorImpl.java:952)
	at com.google.inject.internal.InjectorImpl.getMembersInjector(InjectorImpl.java:957)
	at com.google.inject.internal.InjectorImpl.injectMembers(InjectorImpl.java:943)
	at com.freedomotic.api.Plugin.<init>(Plugin.java:101)
	at com.freedomotic.api.Protocol.<init>(Protocol.java:52)
	at com.freedomotic.plugins.devices.restapiv3.RestAPIv3.<init>(RestAPIv3.java:68)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at com.freedomotic.plugins.impl.BoundleLoaderDevices.loadPlugin(BoundleLoaderDevices.java:93)
	at com.freedomotic.plugins.impl.BoundleLoaderDevices.loadBoundle(BoundleLoaderDevices.java:65)
	... 4 more
`

**Comment by dmitraver** [comment]
@mcicolella hmm, yes I did and all the tests passed in the module, let me try again

**Comment by dmitraver** [comment]
Ok, I see what the problem is, whould fix it asap

**Comment by dmitraver** [comment]
@mcicolella issues should be fixed, plz try it out.

**Comment by mcicolella** [comment]
@dmitraver no more errors previous reported but the websocket doesn't work.
I tried to connect (as in the attached screenshot) but I don't view any message  when I click on a light on floormap.
In the screenshot you can see what type of message I'd receive.

schermata del 2018-10-03 16-29-03

**Comment by dmitraver** [comment]
@mcicolella thanks for checking this one out, I'm not really familiar with a codebase but I would try to check what might have changed between the versions, its actually weird that the behaviour changed as I just replaced the call to the method that no longer exists with an alternative that they suggested in a migration guide...

**Comment by dmitraver** [comment]
@mcicolella I found the solution, was able to test it with wscat as well. Plz give it a try.

**Comment by mcicolella** [comment]
@dmitraver you are amazing!! It works, thanks.
